src/bundle.c

```diff
@@ -723,6 +728,35 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	to_install_files = filter_out_existing_files(to_install_files);
 	grabtime_stop(&times);
 
+	/* Step 1.5: Now that we know the bundle is valid, check if we have enough space */
+	ptr = list_head(to_install_bundles);
+	while (ptr) {
+		subman = ptr->data;
+		ptr = ptr->next;
+
+		struct bundle_result *bundle = NULL;
+		bundle = calloc(sizeof(struct bundle_result), 1);
+		/* copy relevant information over for future use */
+		strncpy(bundle->bundle_name, subman->component, BUNDLE_NAME_MAXLEN - 1);
+		bundle->topsize = subman->contentsize;
+		/* do a deep copy of the includes list */
+		bundle->includes = list_deep_clone_strs(subman->includes);
```

Comment:
I have a really strong feeling that you don't need to duplicate that list to calculate the installation sizes.

Reply:
Admittedly, this a turn-key function from search.c - I think it can be improved and will look into it for V2.